src/js/client/renderer.js

```diff
@@ -221,19 +93,59 @@
             "cacheTemplates": {
                 funcName: "gpii.handlebars.renderer.serverAware.cacheTemplates",
                 args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
-            },
-            "retrieveTemplates": {
-                funcName: "gpii.handlebars.renderer.serverAware.retrieveTemplates",
-                args: ["{that}", "{arguments}.0"]
             }
         },
         events: {
             "onTemplatesLoaded": null
         },
         listeners: {
             "onCreate.loadTemplates": {
-                func: "{that}.retrieveTemplates"
+                funcName: "gpii.handlebars.renderer.serverAware.retrieveResource",
+                args: ["{that}.options.templateUrl", "{that}.cacheTemplates"] // url, callback
+            }
+        }
+    });
+
+    fluid.registerNamespace("gpii.handlebars.renderer.serverMessageAware");
+
+    /**
+     *
+     * The message bundling middleware delivers all available languages.  We look at our current locale and save the
+     * current bundle to the `messages` model variable.
+     *
+     * @param {Object} that - The renderer component itself.
+     * @param {Object} data - The message bundle for the locale as returned by the server.
+     */
+    gpii.handlebars.renderer.serverMessageAware.cacheMessages = function (that, data) {
+        that.applier.change("messages", data);
```

Comment:
For such a lot of structured data, might be safer to operate a transaction which deletes the old messages and then applies the new ones, otherwise there is a risk of stale messages being left behind as mouse droppings if the newer bundle doesn't contain them all

Reply:
Done.